clrdefinitions.cmake

```diff
@@ -31,6 +31,12 @@ elseif (CLR_CMAKE_TARGET_ARCH_ARM64)
 elseif (CLR_CMAKE_TARGET_ARCH_ARM)
   if (CLR_CMAKE_PLATFORM_UNIX)
     add_definitions(-DDBG_TARGET_ARM_UNIX)
+  elseif (WIN32 AND NOT DEFINED CLR_CROSS_COMPONENTS_BUILD)
+    # Set this to ensure we can use Arm SDK for Desktop binary linkage when doing native (Arm32) build
+    add_definitions(-D_ARM_WINAPI_PARTITION_DESKTOP_SDK_AVAILABLE=1)
+    add_definitions(-D_ARM_)
+    add_definitions(-DARM)
+    add_definitions(-D_ARM_WORKAROUND_)
```

Comment:
_ARM_ & ARM are already present at https://github.com/dotnet/coreclr/blob/master/CMakeLists.txt#L386
rest of them can be added there as well as we need to pivot these on host_arch and not target_arch